static/app/views/insights/browser/webVitals/components/pageOverviewSidebar.tsx

```diff
@@ -127,12 +145,49 @@ export function PageOverviewSidebar({
   const ringSegmentColors = theme.chart.getColorPalette(4);
   const ringBackgroundColors = ringSegmentColors.map(color => `${color}50`);
 
-  const {data: issues, isLoading: isLoadingIssues} = useWebVitalsIssuesQuery({
+  const {data: issues} = useWebVitalsIssuesQuery({
     issueTypes: [IssueType.WEB_VITALS],
     transaction,
     enabled: hasSeerWebVitalsSuggestions,
+    // We only poll for issues if we've created them in this session, otherwise we only attempt to load any existing issues once
+    pollInterval: POLL_INTERVAL,
+    eventIds: newlyCreatedIssueEventIds,
   });
 
+  // Creates a new issue for each web vital that has a score under 90 and runs seer autofix for each of them
+  // TODO: Add logic to actually initiate running autofix for each issue. Right now we rely on the project config to automatically run autofix for each issue.
+  const runSeerAnalysis = useCallback(async () => {
```

Comment:
Would it be possible to encapsulate this `runSeerAnalysis` and all of the loading state into another hook? I'm just wondering if it would be cleaner to take all of the stuff that's required to power the seer analysis button and have that logic elsewhere, and this component won't have to define this big callback amongst its rendering code.

Reply:
moved `runSeerAnalysis` to it's own hook. I'll see if I can consolidate more into a hook in a future pr.